# Εκτίμηση ενός Συστήματος Ζήτησης Οικιακής Ενέργειας με Φαινομενικά Ασυσχέτιστη Παλινδρόμηση

## Σύνοψη

Μια περιφερειακή επιχείρηση κοινής ωφέλειας εκτιμά ένα **σύστημα ζήτησης οικιακής ενέργειας** δύο εξισώσεων — τα μερίδια προϋπολογισμού ηλεκτρικής ενέργειας και φυσικού αερίου ως συναρτήσεις της ίδιας τιμής, της διασταυρούμενης τιμής και του εισοδήματος του νοικοκυριού — χρησιμοποιώντας το **PROC SYSLIN** με τη μέθοδο της φαινομενικά ασυσχέτιστης παλινδρόμησης (SUR). Οι δύο εξισώσεις μεριδίων μοιράζονται έναν κοινό προϋπολογισμό νοικοκυριού, οπότε τα σφάλματά τους είναι διασταυρούμενα συσχετισμένα· η SUR εκτιμά τις εξισώσεις από κοινού ώστε να αξιοποιήσει αυτή τη συσχέτιση, ανακτώντας συνεπείς ως προς το πρόσημο επιδράσεις ίδιας και διασταυρούμενης τιμής και παρέχοντας τον διασταυρούμενο πίνακα συνδιακύμανσης και τους ελέγχους περιορισμών που χρειάζεται ένας αναλυτής ζήτησης.

## Πηγές Δεδομένων

| Σύνολο δεδομένων | Γραμμές | Κόκκος | Βασικές μεταβλητές | Περιγραφή |
|---------|------|-------|---------------|-------------|
| `energy` | 60 | μία γραμμή ανά μηνιαία παρατήρηση αγοράς | `month`, `lp_elec`, `lp_gas`, `lincome`, `w_elec`, `w_gas` | Συνθετικός μηνιαίος πίνακας αγοράς οικιακής ενέργειας. Τα `lp_elec`/`lp_gas` είναι οι λογάριθμοι των πραγματικών τιμών ηλεκτρικής ενέργειας και φυσικού αερίου· το `lincome` είναι ο λογάριθμος του πραγματικού εισοδήματος του νοικοκυριού· τα `w_elec`/`w_gas` είναι τα μερίδια προϋπολογισμού δαπανών για ηλεκτρική ενέργεια και φυσικό αέριο, που παράγονται από μια γνωστή δομή ζήτησης τύπου AIDS συν συσχετισμένο διασταυρούμενο θόρυβο. |

# Εκτίμηση ενός Συστήματος Ζήτησης Οικιακής Ενέργειας με Φαινομενικά Ασυσχέτιστη Παλινδρόμηση

Μια περιφερειακή επιχείρηση κοινής ωφέλειας φυσικού αερίου και ηλεκτρισμού θέλει να κατανοήσει πώς τα νοικοκυριά ανακατανέμουν τον ενεργειακό τους προϋπολογισμό μεταξύ **ηλεκτρικής ενέργειας** και **φυσικού αερίου** καθώς μεταβάλλονται οι σχετικές τιμές και το εισόδημα. Το φυσικό πλαίσιο είναι ένα **σύστημα ζήτησης**: ένα σύνολο εξισώσεων μεριδίου προϋπολογισμού που εκτιμώνται από κοινού.

Δύο χαρακτηριστικά καθιστούν την από κοινού εκτίμηση το σωστό εργαλείο:

- Οι εξισώσεις μεριδίου αντλούν από έναν κοινό προϋπολογισμό νοικοκυριού, οπότε τα σφάλματά τους είναι **διασταυρούμενα συσχετισμένα** — όταν ένα νοικοκυριό ξοδεύει περισσότερα σε ηλεκτρική ενέργεια ξοδεύει λιγότερα σε αέριο. Η από κοινού εκτίμηση των εξισώσεων με **φαινομενικά ασυσχέτιστη παλινδρόμηση (SUR)** χρησιμοποιεί αυτή τη συσχέτιση για αποδοτικότητα.
- Η οικονομική θεωρία επιβάλλει **γραμμικούς περιορισμούς** (άθροισμα μεριδίων, ομογένεια, συμμετρία) που ένας εκτιμητής συστήματος μπορεί να επιβάλει άμεσα.

Το `PROC SYSLIN` με την επιλογή `SUR` είναι φτιαγμένο ακριβώς για αυτή την περίπτωση. Προσαρμόζει κάθε εξίσωση μεριδίου, εκτιμά τη διασταυρούμενη συνδιακύμανση σφαλμάτων και επαναπροσαρμόζει το σύστημα με αυτή τη συνδιακύμανση ώστε να παραδώσει αποδοτικές, συνεπείς με τη θεωρία ελαστικότητες — μαζί με τον διασταυρούμενο πίνακα συνδιακύμανσης και τους από κοινού ελέγχους περιορισμών.

Σε αυτό το σημειωματάριο:

1. Παράγουμε έναν συνθετικό μηνιαίο πίνακα αγοράς ενέργειας από μια γνωστή δομή μεριδίων.
2. Προσαρμόζουμε ένα σύστημα μεριδίων δύο εξισώσεων με SUR.
3. Συγκρίνουμε την από κοινού προσαρμογή SUR με την OLS εξίσωση προς εξίσωση.
4. Επιβάλλουμε έναν περιορισμό ομογένειας και διαβάζουμε τον από κοινού F-έλεγχό του.
5. Εξάγουμε τον πίνακα συντελεστών για την αναφορά ελαστικοτήτων.

## Βήμα 1 — Παραγωγή ενός συνθετικού πίνακα αγοράς ενέργειας

Προσομοιώνουμε 60 μηνιαίες παρατηρήσεις ενός μικρού **συστήματος ζήτησης ενέργειας δύο αγαθών** (ηλεκτρική ενέργεια και φυσικό αέριο). Η διαδικασία παραγωγής δεδομένων είναι ένα γραμμικοποιημένο μοντέλο μεριδίου προϋπολογισμού τύπου AIDS:

```
w_elec = a1 + g11*lp_elec + g12*lp_gas + b1*lincome + e1
w_gas  = a2 + g21*lp_elec + g22*lp_gas + b2*lincome + e2
```

Ενσωματώνουμε σκόπιμα **διασταυρούμενη συσχέτιση**: όταν τα νοικοκυριά ξοδεύουν περισσότερα σε ηλεκτρική ενέργεια ξοδεύουν λιγότερα σε αέριο, οπότε τα `e1` και `e2` είναι αρνητικά συσχετισμένα. Ένας κοινός παράγοντας τιμής της αγοράς ενέργειας οδηγεί επίσης και τους δύο λογαρίθμους τιμών μαζί. Αυτά είναι τα χαρακτηριστικά που ανταμείβουν την από κοινού εκτίμηση SUR έναντι της προσαρμογής κάθε εξίσωσης μόνης της.

In [1]:
ΔΕΔΟΜΕΝΑ energy;
   CALL streaminit(70123);

   /* True structural coefficients (linearized AIDS share system) */
   a1  = 0.55;  g11 =  0.12; g12 = -0.08; b1 = -0.030;
   a2  = 0.45;  g21 = -0.08; g22 =  0.10; b2 = -0.025;

   ΕΠΑΝΑΛΗΨΗ month = 1 ΕΩΣ 60;
      /* A common energy-market price factor drives BOTH prices,
         creating the collinearity that makes the problem ill-posed. */
      energy_factor = rand('NORMAL', 0, 0.15);

      lp_elec = 4.6 + energy_factor + rand('NORMAL', 0, 0.05);
      lp_gas  = 4.2 + energy_factor + rand('NORMAL', 0, 0.05);
      lincome = 10.8 + 0.004*month + rand('NORMAL', 0, 0.06);

      /* Negatively correlated cross-equation errors (budget tradeoff) */
      shock = rand('NORMAL', 0, 0.010);
      e1 =  shock + rand('NORMAL', 0, 0.006);
      e2 = -shock + rand('NORMAL', 0, 0.006);

      w_elec = a1 + g11*lp_elec + g12*lp_gas + b1*lincome + e1;
      w_gas  = a2 + g21*lp_elec + g22*lp_gas + b2*lincome + e2;

      ΕΞΟΔΟΣ;
   ΤΕΛΟΣ;

   ΚΡΑΤΗΣΗ month lp_elec lp_gas lincome w_elec w_gas;
ΕΚΤΕΛΕΣΗ;

ΔΙΑΔΙΚΑΣΙΑ ΜΕΣΟΙ ΔΕΔΟΜΕΝΑ=energy n mean std MIN MAX maxdec=3;
   ΜΕΤΑΒΛΗΤΗ w_elec w_gas lp_elec lp_gas lincome;
   ΕΤΙΚΕΤΑ w_elec="Μερίδιο ηλεκτρικής ενέργειας"
         w_gas="Μερίδιο φυσικού αερίου"
         lp_elec="Λογ. τιμής ηλεκτρικής ενέργειας"
         lp_gas="Λογ. τιμής φυσικού αερίου"
         lincome="Λογ. πραγματικού εισοδήματος";
ΕΚΤΕΛΕΣΗ;

                                                  The MEANS Procedure

 Variable  Label                                                             N           Mean     Std Dev     Minimum     Maximum
 --------------------------------------------------------------------------------------------------------------------------------
 w_elec    Μερίδιο ηλεκτρικής ενέργειας                                     60          0.440       0.011       0.417       0.464
 w_gas     Μερίδιο φυσικού αερίου                                           60          0.228       0.014       0.198       0.256
 lp_elec   Λογ. τιμής ηλεκτρικής ενέργειας                                  60          4.617       0.142       4.354       4.911
 lp_gas    Λογ. τιμής φυσικού αερίου                                        60          4.211       0.162       3.818       4.566
 lincome   Λογ. πραγματικού εισοδήματος                                     60         10.916       0.088      10.723      11.174
 ------------------


NOTE: DATA energy


NOTE: Wrote energy (60 rows, 6 columns).
NOTE: DATA elapsed:
  wall  0.01 seconds
  cpu   0.01 seconds
NOTE: PROC MEANS
NOTE: PROC MEANS statement used.


## Βήμα 2 — Βασική εκτίμηση SUR του συστήματος ζήτησης

Εκτιμούμε και τις δύο εξισώσεις μεριδίου από κοινού. Η επιλογή `SUR` λέει στο `PROC SYSLIN` να εκτιμήσει τη διασταυρούμενη συνδιακύμανση σφαλμάτων και να τη χρησιμοποιήσει για μια αποδοτική εφικτή GLS προσαρμογή. Δύο επισημασμένες δηλώσεις `MODEL` (`elec` και `gas`) ορίζουν το σύστημα· κάθε μία παλινδρομεί ένα μερίδιο προϋπολογισμού πάνω στους δύο λογαρίθμους τιμών και τον λογάριθμο του εισοδήματος.

Το SYSLIN αναφέρει τις εκτιμήσεις παραμέτρων κάθε εξίσωσης και, στο τέλος, τον **Διασταυρούμενο Πίνακα Συνδιακύμανσης Μοντέλων (Cross-Model Covariance Matrix)** — την εκτιμώμενη συνδιακύμανση των σφαλμάτων των δύο εξισώσεων που αξιοποιεί η SUR.

In [2]:
ΔΙΑΔΙΚΑΣΙΑ syslin ΔΕΔΟΜΕΝΑ=energy ON;
   elec: ΜΟΝΤΕΛΟ w_elec = lp_elec lp_gas lincome;
   gas:  ΜΟΝΤΕΛΟ w_gas  = lp_elec lp_gas lincome;
ΕΚΤΕΛΕΣΗ;


                       The SYSLIN Procedure

                  SUR Estimation

  Model elec Dependent Variable: w_elec

  Number of Observations                       60
  SSE                                      0.0048
  MSE                                    0.000085
  R-Square                                 0.3865
  Adj R-Sq                                 0.3537

                      Parameter Estimates

                                   Standard
  Variable     DF     Estimate        Error    t Value    Pr > |t|
  ----------  ---  ------------  ----------  ---------  ----------
  Intercept      1      0.801017    0.150270       5.33      0.0000
  LP_ELEC        1      0.112463    0.021244       5.29      0.0000
  LP_GAS         1     -0.097938    0.018646      -5.25      0.0000
  LINCOME        1     -0.042842    0.013238      -3.24      0.0020

  Model gas Dependent Variable: w_gas

  Number of Observations                       60
  SSE                                      0.


NOTE: PROC SYSLIN data=energy

NOTE: Using Python for simultaneous equations estimation (SUR)
NOTE: PROC SYSLIN completed.


## Βήμα 3 — Σύγκριση με OLS εξίσωση προς εξίσωση

Για να δούμε τι μας προσφέρει η SUR, επαναπροσαρμόζουμε τις ίδιες δύο εξισώσεις με τη μέθοδο των ελαχίστων τετραγώνων (OLS, την προεπιλεγμένη μέθοδο, μία εξίσωση τη φορά). Η OLS αγνοεί τη διασταυρούμενη συσχέτιση σφαλμάτων, οπότε είναι συνεπής αλλά λιγότερο αποδοτική από την SUR όταν αυτή η συσχέτιση είναι μη μηδενική — όπως συμβαίνει εδώ εκ κατασκευής.

Η σύγκριση των δύο πινάκων συντελεστών δείχνει πώς οι εκτιμήσεις και τα τυπικά σφάλματά τους μετατοπίζονται μόλις ληφθεί υπόψη η δομή του συστήματος.

In [3]:
ΔΙΑΔΙΚΑΣΙΑ syslin ΔΕΔΟΜΕΝΑ=energy ols;
   elec: ΜΟΝΤΕΛΟ w_elec = lp_elec lp_gas lincome;
   gas:  ΜΟΝΤΕΛΟ w_gas  = lp_elec lp_gas lincome;
ΕΚΤΕΛΕΣΗ;


                       The SYSLIN Procedure

                  OLS Estimation

  Model elec Dependent Variable: w_elec

  Number of Observations                       60
  SSE                                      0.0048
  MSE                                    0.000085
  R-Square                                 0.3865
  Adj R-Sq                                 0.3537

                      Parameter Estimates

                                   Standard
  Variable     DF     Estimate        Error    t Value    Pr > |t|
  ----------  ---  ------------  ----------  ---------  ----------
  Intercept      1      0.801017    0.155544       5.15      0.0000
  LP_ELEC        1      0.112463    0.021989       5.11      0.0000
  LP_GAS         1     -0.097938    0.019300      -5.07      0.0000
  LINCOME        1     -0.042842    0.013702      -3.13      0.0028

  Model gas Dependent Variable: w_gas

  Number of Observations                       60
  SSE                                      0.


NOTE: PROC SYSLIN data=energy

NOTE: Using Python for simultaneous equations estimation (OLS)
NOTE: PROC SYSLIN completed.


## Βήμα 4 — Επιβολή της οικονομικής θεωρίας και έλεγχός της

Η θεωρία ζήτησης λέει ότι οι ονομαστικές επιδράσεις τιμής/εισοδήματος πρέπει να υπακούουν στην **ομογένεια μηδενικού βαθμού**: η κλιμάκωση όλων των τιμών και του εισοδήματος αφήνει την πραγματική ζήτηση αμετάβλητη, οπότε οι συντελεστές τιμής και εισοδήματος εντός μιας εξίσωσης αθροίζονται στο μηδέν. Την επιβάλλουμε στο μερίδιο ηλεκτρικής ενέργειας με μια δήλωση `RESTRICT`.

Το SYSLIN επαναπροσαρμόζει το σύστημα υπό τον περιορισμό και αναφέρει έναν F-έλεγχο **Αποτελεσμάτων Περιορισμού (Restriction Results)** για το αν ο περιορισμός είναι συμβατός με τα δεδομένα — έναν άμεσο, από κοινού έλεγχο της υπόθεσης ομογένειας.

In [4]:
ΔΙΑΔΙΚΑΣΙΑ syslin ΔΕΔΟΜΕΝΑ=energy ON;
   elec: ΜΟΝΤΕΛΟ w_elec = lp_elec lp_gas lincome;
   gas:  ΜΟΝΤΕΛΟ w_gas  = lp_elec lp_gas lincome;

   /* Homogeneity on the electricity share equation:
      price and income coefficients sum to zero. */
   restrict lp_elec + lp_gas + lincome = 0;
ΕΚΤΕΛΕΣΗ;


                       The SYSLIN Procedure

                  SUR Estimation

  Model elec Dependent Variable: w_elec

  Number of Observations                       60
  SSE                                      0.0048
  MSE                                    0.000085
  R-Square                                 0.3865
  Adj R-Sq                                 0.3537

                      Parameter Estimates

                                   Standard
  Variable     DF     Estimate        Error    t Value    Pr > |t|
  ----------  ---  ------------  ----------  ---------  ----------
  Intercept      1      0.801017    0.150270       5.33      0.0000
  LP_ELEC        1      0.112463    0.021244       5.29      0.0000
  LP_GAS         1     -0.097938    0.018646      -5.25      0.0000
  LINCOME        1     -0.042842    0.013238      -3.24      0.0020

  Model gas Dependent Variable: w_gas

  Number of Observations                       60
  SSE                                      0.


NOTE: PROC SYSLIN data=energy

NOTE: Using Python for simultaneous equations estimation (SUR)
NOTE: PROC SYSLIN completed.


## Βήμα 5 — Καταγραφή εκτιμήσεων για την αναφορά ελαστικοτήτων

Για μια τελική αναφορά διατηρούμε τους συγκλίνοντες συντελεστές με το `OUTEST=`. Το προκύπτον σύνολο δεδομένων φέρει μία γραμμή ανά εξίσωση με τις εκτιμήσεις του σταθερού όρου και των κλίσεων συν στατιστικά προσαρμογής, τα οποία τροφοδοτούν τους δικούς της υπολογισμούς ελαστικότητας ίδιας και διασταυρούμενης τιμής της επιχείρησης στα μέσα μερίδια του δείγματος. Το εκτυπώνουμε για αρχείο.

In [5]:
ΔΙΑΔΙΚΑΣΙΑ syslin ΔΕΔΟΜΕΝΑ=energy ON outest=demand_est;
   elec: ΜΟΝΤΕΛΟ w_elec = lp_elec lp_gas lincome;
   gas:  ΜΟΝΤΕΛΟ w_gas  = lp_elec lp_gas lincome;
ΕΚΤΕΛΕΣΗ;

ΔΙΑΔΙΚΑΣΙΑ ΕΚΤΥΠΩΣΗ ΔΕΔΟΜΕΝΑ=demand_est noobs;
   TITLE "Εκτιμήσεις συντελεστών συστήματος ζήτησης SUR";
ΕΚΤΕΛΕΣΗ;
TITLE;


                       The SYSLIN Procedure

                  SUR Estimation

  Model elec Dependent Variable: w_elec

  Number of Observations                       60
  SSE                                      0.0048
  MSE                                    0.000085
  R-Square                                 0.3865
  Adj R-Sq                                 0.3537

                      Parameter Estimates

                                   Standard
  Variable     DF     Estimate        Error    t Value    Pr > |t|
  ----------  ---  ------------  ----------  ---------  ----------
  Intercept      1      0.801017    0.150270       5.33      0.0000
  LP_ELEC        1      0.112463    0.021244       5.29      0.0000
  LP_GAS         1     -0.097938    0.018646      -5.25      0.0000
  LINCOME        1     -0.042842    0.013238      -3.24      0.0020

  Model gas Dependent Variable: w_gas

  Number of Observations                       60
  SSE                                      0.


NOTE: PROC SYSLIN data=energy

NOTE: Using Python for simultaneous equations estimation (SUR)
NOTE: Wrote OUTEST= dataset demand_est (2 rows).
NOTE: PROC SYSLIN completed.
NOTE: PROC PRINT data=demand_est

NOTE: PROC PRINT completed: 2 observations printed, 12 variables


## Ερμηνεία των αποτελεσμάτων

**Συνέπεια προσήμου.** Οι εκτιμήσεις SUR ανακτούν τη δομή υποκατάστασης που είναι ενσωματωμένη στα δεδομένα. Στην εξίσωση μεριδίου ηλεκτρικής ενέργειας ο **συντελεστής του ίδιου λογαρίθμου τιμής είναι θετικός** (`LP_ELEC` = 0.112) ενώ ο **συντελεστής της διασταυρούμενης τιμής είναι αρνητικός** (`LP_GAS` = -0.098)· στην εξίσωση μεριδίου αερίου το μοτίβο το αντικατοπτρίζει (ίδιο `LP_GAS` = 0.114, διασταυρούμενο `LP_ELEC` = -0.075). Κάθε επίδραση ίδιας και διασταυρούμενης τιμής είναι στατιστικά σημαντική (κάθε `Pr > |t|` κάτω από 0.005), οπότε τα πρόσημα υποκατάστασης ταυτοποιούνται σταθερά αντί να είναι τεχνουργήματα μιας θορυβώδους προσαρμογής.

**Η SUR αξιοποιεί τη διασταυρούμενη συσχέτιση.** Ο **Διασταυρούμενος Πίνακας Συνδιακύμανσης Μοντέλων** δείχνει ένα αρνητικό στοιχείο εκτός διαγωνίου (-0.000068): τα νοικοκυριά ανταλλάσσουν τη δαπάνη ηλεκτρικής ενέργειας με τη δαπάνη αερίου, ακριβώς όπως κατασκευάστηκε. Επειδή αυτή η συνδιακύμανση είναι μη μηδενική, η από κοινού εκτίμηση SUR είναι πιο αποδοτική από την προσαρμογή κάθε εξίσωσης μόνης της — τα τυπικά σφάλματα SUR στο Βήμα 2 είναι ομοιόμορφα ελαφρώς μικρότερα από τα τυπικά σφάλματα OLS εξίσωση προς εξίσωση στο Βήμα 3 (για παράδειγμα, το τυπικό σφάλμα του `LP_ELEC` του αερίου πέφτει από 0.0264 υπό OLS σε 0.0255 υπό SUR), ενώ οι σημειακές εκτιμήσεις παραμένουν αμετάβλητες. Αυτή η στενότερη στατιστική συμπερασματολογία είναι το όφελος της μοντελοποίησης του συστήματος αντί για δύο ξεχωριστές παλινδρομήσεις.

**Η θεωρία επιβεβαιώνεται.** Η επιβολή **ομογένειας μηδενικού βαθμού** στο μερίδιο ηλεκτρικής ενέργειας (συντελεστές τιμής και εισοδήματος που αθροίζονται στο μηδέν) μέσω του `RESTRICT` παρήγαγε έναν F-έλεγχο Αποτελεσμάτων Περιορισμού 2.14 με `Pr > F` = 0.149. Ο περιορισμός **δεν απορρίπτεται**, οπότε η πρόβλεψη ομογένειας της θεωρίας ζήτησης του καταναλωτή είναι συμβατή με αυτό το δείγμα — η επιχείρηση μπορεί να μεταφέρει τις περιορισμένες, συνεπείς με τη θεωρία εκτιμήσεις στην αναφορά με εμπιστοσύνη.

**Επιχειρησιακή χρήση.** Ο πίνακας `OUTEST=` διατηρεί μία γραμμή ανά εξίσωση με τις εκτιμήσεις του σταθερού όρου και των κλίσεων και τα στατιστικά προσαρμογής. Αυτοί οι συντελεστές μετατρέπονται άμεσα σε ελαστικότητες ίδιας και διασταυρούμενης τιμής και σε ελαστικότητα εισοδήματος (δαπάνης) στα μέσα μερίδια του δείγματος (`W_ELEC` ~ 0.44, `W_GAS` ~ 0.23 από το Βήμα 1), δίνοντας στην επιχείρηση μια αξιόπιστη, συνεπή με τη θεωρία βάση για πρόβλεψη ζήτησης και σενάρια σχεδιασμού τιμολογίων.